File: 14. Longest Common Prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        prefix=''
        if len(strs)==0: return prefix
        for x in zip(*strs):
            if len(set(x)) == 1:
                prefix += x[0]
            else: return prefix
        return prefix

File: 14. Longest Common Prefix.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function (strs) {
    if (strs == null || strs.length == 0) return '';
    for (let i = 0; i < strs[0].length ; i++) {
        let c = strs[0][i];
        for (let j = 1; j < strs.length; j++) {
            if (i == strs[j].length || strs[j][i] != c)
                return strs[0].substring(0, i);
        }
    }
    return strs[0];
};

